.github/workflows/ci.yml

```diff
@@ -40,6 +40,12 @@ jobs:
         uses: actions/checkout@v2.4.0
       - name: Update Rust toolchain
         run: rustup update
+      - name: Install musl-tools
+        run: sudo apt-get install musl-tools --no-install-recommends
+      - name: Install Rust linux-musl target
+        run: rustup target add x86_64-unknown-linux-musl
+      - name: Install pack
+        uses: buildpacks/github-actions/setup-pack@v4.5.0
```

Comment:
I'm not sure what to do with this comment. Merging this will indeed increate CI times by a lot, not only is there a new crate, we'll now actually run buildpacks written with libcnb.rs on CI.

The configuration ideas you mention wouldn't be in this crate, when you say that you're thinking about local dev UX, what did you have in mind as a fix?

Reply:
My main concerns are:
1. Locally, it's useful to have a way to run tests quickly. Should we start recommending people use more args to `cargo test`, or go the feature route so they don't have to?
2. On CI, ideally (a) the unit tests job would be fast, so it can pass/fail fast, giving feedback sooner, (b) the test job runtimes would be balanced (ie: one test not the long pole in terms of total end to end time). Moving the tests added here to the `integration` job instead of the `test` job would help with both (a)+(b), and give others copying from this repo an idea of how to structure their own tests.

If we went the feature attribute route, that would be something in this repo, as would any CI config tweaks.

However I'm happy to resolve these via follow-up GitHub issues, rather than here, unless you have an immediate quick idea as to our preferred way forwards :-)

Reply:
Nope, no quick idea. Sounds like a good topic for the "Testing Buildpacks" chapter of the libcnb.rs book :)

Reply:
Agree (1) is a docs issue, (2) still needs a GitHub issue though. I'd like to undo the CI E2E time regressions from this PR at some point :-)

Reply:
Filed #301 for this.